lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes-dialog.component.ts

```diff
@@ -15,15 +15,45 @@
  * limitations under the License.
  */
 
-import { Component, ViewEncapsulation } from '@angular/core';
+import { Component, Inject, ViewEncapsulation } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogModule } from '@angular/material/dialog';
+import { UnsavedChangesDialogData } from './unsaved-changes-dialog.model';
+import { MatCheckboxChange, MatCheckboxModule } from '@angular/material/checkbox';
+import { ReactiveFormsModule } from '@angular/forms';
+import { TranslateModule } from '@ngx-translate/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MatIconModule } from '@angular/material/icon';
+import { CommonModule } from '@angular/common';
+import { StorageService } from '../../common';
 
 /**
  * Dialog which informs about unsaved changes. Allows discard them and proceed or close dialog and stop proceeding.
+ * Can be customized with data object - UnsavedChangesDialogData.
+ * If data.checkboxText is provided, checkbox will be displayed with the checkbox description.
+ * If data.confirmButtonText is provided, it will be displayed on the confirm button.
+ * If data.headerText is provided, it will be displayed as the header.
+ * If data.descriptionText is provided, it will be displayed as dialog content.
  */
 @Component({
+    standalone: true,
     selector: 'adf-unsaved-changes-dialog',
     encapsulation: ViewEncapsulation.None,
     templateUrl: './unsaved-changes-dialog.component.html',
-    styleUrls: ['./unsaved-changes-dialog.component.scss']
+    styleUrls: ['./unsaved-changes-dialog.component.scss'],
+    host: { class: 'adf-unsaved-changes-dialog' },
+    imports: [MatDialogModule, TranslateModule, MatButtonModule, MatIconModule, CommonModule, MatCheckboxModule, ReactiveFormsModule]
 })
-export class UnsavedChangesDialogComponent {}
+export class UnsavedChangesDialogComponent {
+    private readonly unsavedAiChangesModalVisible = 'unsaved_ai_changes__modal_visible';
```

Comment:
Let's keep it as global variable on the core level so that it can be reused in other repos